app/models/shortener/shortened_url.rb

```diff
@@ -7,7 +7,12 @@ class Shortener::ShortenedUrl < ActiveRecord::Base
   before_create :generate_unique_key
 
   # allows the shortened link to be associated with a user
-  belongs_to :owner, polymorphic: true
+  if ActiveRecord::VERSION::MAJOR >= 5
+    # adds rails 5 compatibility to have nil values as owner
+    belongs_to :owner, polymorphic: true, optional: true
```

Comment:
@craigmcnamara: Given that `belongs_to` accepts and `options` hash in Rails 4 (and presumably Rails 3 and 5), is this conditional required? i.e. if we pass the `optional` param in the options hash to rails4 belongs_to, it would just get ignored right? Which would allow us to remove the conditional for simplicity sake.

https://apidock.com/rails/ActiveRecord/Associations/ClassMethods/belongs_to

Reply:
Unfortunately, according to TravisCI that's not the case: https://travis-ci.org/jpmcgrath/shortener/jobs/408740930

Lots of rails code that takes options also does `Hash#assert_valid_keys` on the options

Reply:
Awesome, thanks for verifying that theory. I will cut a new version of the gem.